python/testSrc/com/jetbrains/python/refactoring/PyMoveTest.java

```diff
@@ -427,6 +427,38 @@ public void testMoveSymbolDoesntCreateInitPyInSourceRoot() {
     });
   }
 
+  //PY-44858
+  public void testMoveNotCreateInitPyForNamespacePackagesToAnotherDirectory() {
+    doComparingDirectories(testDir -> {
+      final VirtualFile sourceRoot = testDir.findFileByRelativePath("src");
+      runWithSourceRoots(Collections.singletonList(sourceRoot), () -> moveSymbols(testDir, "src/pkg/subpkg/B/module_b.py", "myfunc"));
+    });
+  }
+
+  //PY-44858
+  public void testMoveNotCreateInitPyForNamespacePackagesInSameDirectory() {
+    doComparingDirectories(testDir -> {
+      final VirtualFile sourceRoot = testDir.findFileByRelativePath("src");
+      runWithSourceRoots(Collections.singletonList(sourceRoot), () -> moveSymbols(testDir, "src/pkg/subpkg/module_b.py", "myfunc"));
+    });
+  }
+
+  //PY-44858
+  public void testMoveNotCreateInitPyForNamespacePackagesToParentDirectory() {
+    doComparingDirectories(testDir -> {
+      final VirtualFile sourceRoot = testDir.findFileByRelativePath("src");
+      runWithSourceRoots(Collections.singletonList(sourceRoot), () -> moveSymbols(testDir, "src/pkg/subpkg/B/module_b.py", "myfunc"));
+    });
+  }
+
+  //PY-44858
+  public void testMoveNotCreateInitPyForNamespacePackagesToChildDirectory() {
```

Comment:
For this kind of test scenarios you don't need namespace packages to be inside an explicit source root. So inside the test data intermediate "src" root could be removed and the test code could be simplified to:
```java
//PY-44858
public void testMoveNotCreateInitPyForNamespacePackagesToChildDirectory() {
  doMoveSymbolsTest("pkg/subpkg/A/B/module_b.py", "myfunc");
}
```
I mean the following "flattened" hierarchy in test data:
```
community/python/testData/refactoring/move/moveNotCreateInitPyForNamespacePackagesToChildDirectory/
├── after
│   └── src
│       └── pkg
│           └── subpkg
│               └── A
│                   ├── B
│                   │   └── module_b.py
│                   └── module_a.py
└── before
    └── src
        └── pkg
            └── subpkg
                └── A
                    └── module_a.py

```